### PR TITLE
Conditionally display filter reset button.

### PR DESCRIPTION
**Pull request by jameskoster**
## What?
Only display the filter reset button when:

1. Filters are applied
2. The button is focussed

## Why?
When no filters are added the button adds noise to the UI. In terms of discoverability it's arguably more helpful to reveal the button when it's actionable.   

## How?
Adjust the opacity of the button when focus is within `.dataviews-filters__reset-button-container`.

## Testing Instructions
1. Open a data view like Manage all pages.
2. Observe no reset button.
3. Tab to the reset button and notice that it appears, disabled.
4. Add filters, notice the reset button appears and can be actioned.
5. Reset the filters and notice that focus remains on the disabled button, and that it remains visible until focus moves.

https://github.com/WordPress/gutenberg/assets/846565/de80ec7d-ee29-4b84-8d7d-b9976adfecb6

### CODE REVIEW

File: packages/dataviews/src/reset-filters.js [hunk 1]
```diff
@@ -16,21 +16,24 @@ export default function ResetFilter( { filters, view, onChangeView } ) {
...
-					...view,
-					page: 1,
-					search: '',
-					filters: [],
-				} );
-			} }
-		>
-			{ __( 'Reset filters' ) }
-		</Button>
+		<div className="dataviews-filters__reset-button-container">
```
Comment (andrewhayward) [inline packages/dataviews/src/reset-filters.js]: I'm guessing this extra `<div>` is here for specificity purposes? If so, we could use an existing wrapper instead...

```scss
.dataviews-filters__container {
  ...

  .dataviews-filters__reset-button[aria-disabled="true"] {
    &,
    &:hover {
      opacity: 0;
    }

    &:focus {
      opacity: 1;
    }
  }
}
```
...then this file doesn't have to change at all. Minor point though, so not a blocker.

**Review by andrewhayward — APPROVED** [review]
> ...it's arguably more helpful to reveal the button when it's actionable.

One of those things that ideally we'd run through usability testing, but 🤷‍♂️ From an implementation perspective though, this works for me (see note on redundant wrappers) 🚀

File: packages/dataviews/src/reset-filters.js (hunk 1 above)
Comment (jameskoster) [reply packages/dataviews/src/reset-filters.js]: Good point, I'll make this change.

**Comment by jasmussen** [comment]
Seems worth a try!

**Comment by jameskoster** [comment]
Added the Backport label, but it's not _essential_ so feel free to omit if necessary.

**Comment by youknowriad** [comment]
Since we're passed RC1, I'm actually removing all non essential changes.

**Pull request merged**